### PR TITLE
CRM-17620 - Cleanup and abstract expandRow js

### PR DESCRIPTION
**Pull request by colemanw**
https://issues.civicrm.org/jira/browse/CRM-17620

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Now you just have to police that none of the other 3+ variants gets copy-pasted next time someone wants an expander :-)

File: templates/CRM/Contribute/Form/Selector.tpl [hunk 1]
```diff
@@ -65,8 +65,9 @@
     <td>{$row.contact_type}</td>
       <td><a href="{crmURL p='civicrm/contact/view' q="reset=1&cid=`$row.contact_id`"}">{$row.sort_name}</a></td>
     {/if}
-    {assign var="targetRowID" value="paymentDetails"|cat:$row.contribution_id}
-    {include file='CRM/common/expandRow.tpl' rowEntityID=$row.contribution_id rowEntity='contribution' targetRowID=$targetRowID baseUrl='payment'}
+    <td>
+      <a class="crm-expand-row" href="{crmURL p='civicrm/payment' q="view=transaction&component=contribution&action=browse&cid=`$row.contact_id`&id=`$row.contribution_id`&selector=1"}"></a>
+    </td>
```
Comment (colemanw) [inline templates/CRM/Contribute/Form/Selector.tpl]: I kept with your markup here @eileenmcnaughton but I'm not sure the expander belongs in its own `td` - what's wrong with just sticking it in front of the amount?

Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Selector.tpl]: I didn't start from scratch. I started from what was happening on the pledge tab & took that as my starting point. I don't have a problem with the change you suggest - other than that is should potentially be consistent with pledge tab.

NB just to mention it so at least I've said it - I feel like memberships & events would ideally expand to show contributions and/or membership log lines but in my thinking not line items. Although they are linked to memberships & events in the table I feel we should encourage people to conceptualise  line items as sub entities on contributions

**Comment by colemanw** [comment]
@eileenmcnaughton I think I might have got them all... are there others you noticed?

**Comment by eileenmcnaughton** [comment]
Only the manage groups one - but I don't know if that is similar enough to be included

**Comment by colemanw** [comment]
No, that one's crazy.

**Pull request merged**